rxandroidble/src/main/java/com/polidea/rxandroidble/exceptions/BleIllegalOperationException.java

```diff
@@ -1,11 +1,24 @@
 package com.polidea.rxandroidble.exceptions;
 
+import java.util.UUID;
+
 /**
- *
+ * This exception is thrown when a non-supported operation has been requested upon a characteristic, eg. write operation on a
+ * characteristic with only read property.
  */
 public class BleIllegalOperationException extends RuntimeException {
 
-    public BleIllegalOperationException(String message) {
+    public final UUID characteristicUUID;
+    public final int supportedProperties;
+    public final int neededProperties;
+
+    public BleIllegalOperationException(String message,
```

Comment:
Maybe we could annotate the ints with `@BluetoothGattCharacteristicProperty` to make it more precise what is this `int` related to. `BluetoothGattCharacteristicProperty` would then be put on the public API.